### PR TITLE
added_case_list_in_beneficiary_view_page

### LINKED ISSUE #20: Add case list in beneficiary view page
We would like to see the cases that were created for the beneficiary in the beneficiary  view page. 

please add list of cases and its current status in the beneficiary view page.

### PR DESCRIPTION
**Pull request by Mouroof**

### CODE REVIEW

File: cms/use_cases/case.py [hunk 1]
```diff
@@ -74,5 +74,12 @@ def view_case(repo, case_id):
         if case is None:
             raise Exception("Case not found")
         return ResponseSuccess(case)
+    except Exception as exc:
+        return ResponseFailure(ResponseTypes.SYSTEM_ERROR, exc)
+
+def case_list_from_beneficiary_id(repo, beneficiary_id):
+    try:
+        case = repo.case_list_with_beneficiaryid(beneficiary_id)
```
Comment (ik-nitk) [inline cms/use_cases/case.py]: name the variables which gives actual sense.  You can name it case_list

File: web/backend_api_builder.py [hunk 2]
```diff
@@ -70,3 +70,11 @@ def case_id(self, case_id):
...
         """
         return self.cases + '/{}'.format(case_id)
+
+    def case_list(self,beneficiary_id):
+        """
+        Generates endpoint for retrieving/updating task with given cases_id
+        :param cases_id: unique cases id
+        :return: case endpoint
+        """
+        return self.cases + '/caselistfrombenificiaryid'+'/{}'.format(beneficiary_id)
```
Comment (ik-nitk) [inline web/backend_api_builder.py]: change this to :  
self.cases + '/beneficiery_id?beneficiery_id={}'.format(beneficiery_id)

File: web/templates/beneficiaries/view.html [hunk 3]
```diff
@@ -100,5 +101,42 @@
...
+            </div>
+            <table
+                id="data"
+                class="table table-striped table-sm mx-auto text-center"
+            >
+                <thead>
+                    <tr>
+                        <th>ID</th>
+                        <th>title</th>
+                        <th>Purpose</th>
```
Comment (ik-nitk) [inline web/templates/beneficiaries/view.html]: Don't want purpose / contact details .. it will increase size of row.   just link/title/state is enough

**Review by ik-nitk — CHANGES_REQUESTED** [review]
can you please address the minor comments.

File: web/backend_api_builder.py (hunk 2 above)
Comment (Mouroof) [reply web/backend_api_builder.py]: ok

File: cms/use_cases/case.py (hunk 1 above)
Comment (Mouroof) [reply cms/use_cases/case.py]: when I name it case_list it gives error as there is already a method called case_list() in case.py file.

Comment (ik-nitk) [reply cms/use_cases/case.py]: you can rename it to different thing like clist or anything else ?

Comment (Mouroof) [reply cms/use_cases/case.py]: Ok

File: application/rest/case.py [hunk 4]
```diff
@@ -95,4 +96,13 @@ def create_case():
         status=STATUS_CODES[response.type],
     )
 
+@blueprint.route("/api/v1/cases/<id>/beneficiary_id", methods=["GET"])
```
Comment (ik-nitk) [inline application/rest/case.py:99]: this is wrong .... it should be 

/api/v1/cases?benerificiary_id=<id>

File: web/backend_api_builder.py [hunk 5]
```diff
@@ -70,3 +70,11 @@ def case_id(self, case_id):
...
         """
         return self.cases + '/{}'.format(case_id)
+
+    def case_list(self,beneficiary_id):
+        """
+        Generates endpoint for retrieving/updating task with given cases_id
+        :param cases_id: unique cases id
+        :return: case endpoint
+        """
+        return self.case_id(beneficiary_id) + '/beneficiary_id?beneficiary_id={}'.format(beneficiary_id)
```
Comment (ik-nitk) [inline web/backend_api_builder.py]: this is also wrong .. the final api should be 
cases?beneficiary_id?beneficiary_id={}

**Review by ik-nitk — CHANGES_REQUESTED** [review]
please make changes

File: cms/use_cases/case.py [hunk 6]
```diff
@@ -74,5 +74,12 @@ def view_case(repo, case_id):
         if case is None:
             raise Exception("Case not found")
         return ResponseSuccess(case)
+    except Exception as exc:
+        return ResponseFailure(ResponseTypes.SYSTEM_ERROR, exc)
+
+def case_clist(repo, beneficiary_id):
```
Comment (ik-nitk) [inline cms/use_cases/case.py:80]: creating 2 functions with same name ?  you can use the existing case_list for this

**Pull request closed without merging**